### ISSUE TITLE
[Feature Request] Allow overriding of worker thread count

### ISSUE THREAD

**Issue by Earthmark** [Neos-Metaverse/NeosPublic#1805]
## Is your feature request related to a problem? Please describe.
Neos seems to spawn a lot of threads, and neos to date is the only application I've used that's gotten my CPU cooler to actually kick on full blast. It's also been reported from multiple sources that higher core counts seem to adversely affect performance, where users with lower core counts seem to get higher performance than users with on paper more powerful machines.

## Relevant issues
None seen at this point.

## Describe the solution you'd like
A command line or config file argument, that when defined provides an alternate estimation of core count, instead of virtual core count.

## Describe alternatives you've considered
Modifying the neos assembly itself, isolating neos to only run on some cores through task manager. However, neos would still create the same number of threads and I have a feeling the sheer amount of context switching the 127+ threads I've seen neos spawn is relevant to the performance issue.

## Additional context
Some of the profiler CPU patterns feel strange, there's a lot of 100% parked threads, and threads that are parked like 99% of the time. It's quite interesting. Either way, thread spam bad!

**Comment by Frooxius** [comment]
It doesn't unfortunately work like that. I could expose thread count for the background workers, but those already match the core count on the system and are used for asynchronous background jobs. You shouldn't really be changing or reducing those, it's not something that the end user should ever worry about and it can cause results to be even worse.

A lot of the threads you don't really have control over. Large number of them are managed by the .NET Runtime (Mono), which has its own threadpool for asynchronous tasks, that it manages and scales based on its own internal mechanics. It's not something you can/should control either. 

A lot of other separate systems have their own threads and thread management, like the crunch library, compressonator, REST API libraries, Unity, libVLC and so on. That's normal and unless those processes are active, those threads won't really have any adverse effect, as they're just idle as you observed.

I'm not fully sure what you're basing the adverse performance with multiple core counts on. There are some issues with heavy multithreading, they mostly have to do with thread starvation, but it's not the main source of the performance issues. Keep in mind that CPU's that have more cores might have weaker single-core performance and that can be cause of bottle-necking instead. Without profiling tools it's hard to tell. I do profiling and various optimizations over time to improve the performance, so that should help as time goes, but there's no single silver bullet for that.

In short, this won't get implemented, sorry, in big part because for a lot of the threads it can't and in another part that it's not the right approach anyways.
